zrml/swaps/src/lib.rs

```diff
@@ -1384,6 +1386,22 @@ mod pallet {
             Ok(next_pool_id)
         }
 
+        fn destroy_pool(pool_id: PoolId) -> Result<Weight, DispatchError> {
+            let pool = Self::pool_by_id(pool_id)?;
+            let pool_account = Self::pool_account_id(pool_id);
+            for asset in pool.assets.into_iter() {
+                let amount = T::Shares::free_balance(asset, &pool_account);
+                T::Shares::slash(asset, &pool_account, amount);
+            }
+            let pool_share_id = Self::pool_shares_id(pool_id);
+            let (_, liquidity_providers) = T::Shares::accounts_by_currency_id(pool_share_id);
+            T::Shares::destroy_all(pool_share_id, liquidity_providers.iter().cloned());
+            Pools::<T>::remove(pool_id);
```

Comment:
For Rikiddo pools the Rikiddo pallet must be informed that the Rikiddo data for that pool should be cleaned up: `T::RikiddoSigmoidFeeMarketEma::destroy(pool_id)?`

Reply:
My plan was to exclude Rikiddo from all this stuff for now (since it's disabled anyways). Is that OK? I could add a TODO marker that mentions that these functions are only to be used with `CPMM` markets.

Reply:
This is fine in my opinion. But in case we go that path, please:
1. Enforce in `admin_destroy_market` that the `ScoringRule` is `CPMM`. The function should ideally just throw an error that indicates that the operation is not supported for `RikiddoSigmoidFeeMarketEma` markets.
2. Create a new issue to implement the support and reference both my review comments in it (because they state what changes are necessary).

In regards to this review, besides the Rikiddo compatibility the only change that I would insist on is [changing the zero weight to something significant](https://github.com/zeitgeistpm/zeitgeist/pull/586#discussion_r871332474) like double-digit weight per block.

Reply:
Ok, done!